druid/src/command.rs

```diff
@@ -15,19 +15,69 @@
 //! Custom commands.
 
 use std::any::Any;
-use std::sync::{Arc, Mutex};
+use std::{
+    marker::PhantomData,
+    sync::{Arc, Mutex},
+};
 
 use crate::{WidgetId, WindowId};
 
+/// An untyped identifier for a `Selector`.
+pub type SelectorSymbol = &'static str;
+
+/// An identifier for a particular command.
+///
+/// This should be a unique string identifier. Certain `Selector`s are defined
+/// by druid, and have special meaning to the framework; these are listed in the
+/// [`druid::commands`] module.
+///
+/// [`druid::commands`]: commands/index.html
+#[derive(Debug, PartialEq, Eq)]
+pub struct Selector<T = ()>(SelectorSymbol, PhantomData<*const T>);
+
+// This has do be done explicitly, to avoid the Copy bound on `T`.
+// See https://doc.rust-lang.org/std/marker/trait.Copy.html#how-can-i-implement-copy .
+impl<T> Copy for Selector<T> {}
```

Comment:
just in general I tend to throw these sorts of impl blocks down at the bottom of the file, for a better reading experience; this is a matter of personal preference though.

Reply:
I do want to reorder a couple things anyways, currently they are just where I could see the context right away.